File: atcoder.jp/contests/abc160/tasks/abc160_b/main.ts
import { readFileSync } from "fs";
let inputLinesIndex = 0;
const inputLines = readFileSync("/dev/stdin", "utf8").split("\n");
const input = () => inputLines[inputLinesIndex++];
const print = (...messages: any) => console.log(...messages);

const X = +input();

const coin500 = Math.floor(X / 500);
const coin5 = Math.floor((X % 500) / 5);

const ans = 1000 * coin500 + 5 * coin5;
print(ans);

File: atcoder.jp/contests/abc160/tasks/abc160_b/main.cpp
#define _GLIBCXX_DEBUG
#include <bits/stdc++.h>
using namespace std;

template <typename T>
void print(T value) {
  cout << value << endl;
}

int main() {
  int X;
  cin >> X;

  int coin500 = X / 500;
  int coin5 = X % 500 / 5;

  int ans = 1000 * coin500 + 5 * coin5;
  print(ans);
}
